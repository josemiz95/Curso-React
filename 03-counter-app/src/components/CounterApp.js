import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({numero}) => {
      const [ value, setValue ] = useState(numero);

      // handleAdd
      const handleAdd = (e) => setValue(value+1);
      const handleSub = (e) => setValue(value-1);
      const handleRest = (e) => setValue(numero);
      
      return (
            <>
                  <h1>Counter App</h1>
                  <h2> {value}</h2>

                  <button onClick={ handleAdd }>+1</button>
                  <button onClick={ handleRest }>Reset</button>
                  <button onClick={ handleSub }>-1</button>
            </>
      );
};

CounterApp.propTypes = {
	numero: PropTypes.number
};

CounterApp.defaultProps = {
	numero: 0
};

export  {
      CounterApp as default
}