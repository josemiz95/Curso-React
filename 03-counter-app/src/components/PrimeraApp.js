import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ nombre, subtitulo }) => {    
    	//Fragment es para evitar poner un div
    	let html = (
		<>
			<h1>Hola {nombre}</h1>
			<p>{subtitulo}</p>
		</>
	 );

	 return html;
}

// Validacion de tipo de objectos
PrimeraApp.propTypes = {
	nombre: PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
	subtitulo: 'Este es el valor por defecto'
};

export {
	PrimeraApp as default
}