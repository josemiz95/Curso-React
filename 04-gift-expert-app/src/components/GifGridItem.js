import React from 'react'

export const GifGridItem = ({title, image}) => {
  return (
    <div className="card animate__animated animate__fadeIn">
        <img src={image} alt={title}/>
    </div>
  )
}
