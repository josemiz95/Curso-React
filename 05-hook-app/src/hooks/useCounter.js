import { useState } from "react"

export const useCounter = (initialState = 1) => {
    const [state, setState] = useState(initialState);

    const operate = num => setState(state+num);
    const reset = num => setState(initialState);

    const increment = (factor=1) => operate(factor);
    const decrement = (factor=1) => operate(-1*factor);

    return {state, increment, decrement, operate, reset};
}
