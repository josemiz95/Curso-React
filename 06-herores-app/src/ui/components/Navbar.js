import { useContext } from 'react';
import { Link, NavLink, useNavigate } from 'react-router-dom';
import { AuthContext } from '../../auth/authContext';
import { types } from '../../types';

export const Navbar = () => {
    const authContext = useContext(AuthContext);

	const navigate = useNavigate();
    const handleLogout = () => {
        authContext.dispatch({ type: types.logout });
        navigate('/login', { replace: true })
    };

    const navItemClass = ({isActive}) =>  "nav-item nav-link " + (isActive?'active':'');

    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link className="navbar-brand" to="/">
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">
                    <NavLink className={navItemClass} to="/marvel">
                        Marvel
                    </NavLink>

                    <NavLink className={navItemClass} to="/dc">
                        DC
                    </NavLink>

                    <NavLink className={navItemClass} to="/search">
                        Search
                    </NavLink>

                    <NavLink className={navItemClass} to="/realtime/search">
                        SearchRealtime
                    </NavLink>
                </div>
            </div>

            <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
                <ul className="navbar-nav ml-auto">
                    <span className="nav-item nav-link text-info">
                        {authContext?.user?.name}
                    </span>
                    <button className="nav-item nav-link btn" to="/login" onClick={handleLogout}>
                        Logout
                    </button>
                </ul>
            </div>
        </nav>
    )
}