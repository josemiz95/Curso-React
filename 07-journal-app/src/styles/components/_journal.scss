.journal__main-content{
	display: flex;
}

.journal__sidebar {
	background-color: $dark-grey;
	color: $white;
	display: flex;
	flex-direction: column;
	height: 100vh;
	padding: 0 1rem;
	width: 450px;
}

.journal__sidebar-navbar{
	display: flex;
	justify-content: space-between;

	h3{
		font-weight: lighter;
		span {
			padding: 0 5px;
		}
	}
}

.journal__new-entry {
	align-items: center;
	cursor: pointer;
	display: flex;
	flex-direction: column;
	justify-content: center;
	margin-top: 30px;
	width: 100%;

	transition: color .3s ease;

	&:hover {
		color: darken($color: white, $amount: 20);
	}
}

.journal__entries {
	flex: 1 1 auto;
	margin-top: 30px;
	overflow-y: auto;
	overflow-x: hidden;
}

.journal__entry{
	background-color: white;
	border-radius: 5px;
	color: $dark-grey;
	display: flex;
	margin-bottom: 10px;
	overflow: hidden;
}

.journal__entry-picture {
	height: 75px;
	width: 75px;
	min-height: 75px;
	min-width: 75px;
}

.journal__entry-content{
	padding: 10px;
}

.journal__entry-title{
	font-size: 20px;
	font-weight: bold;
}

.journal__entry-content{
	font-size: 14px;
}

.journal__entry-date-box{
	align-items: center;
	display: flex;
	flex-direction: column;
	justify-content: center;
	padding: 5px;
	span{
		font-size: 14px;
	}
}