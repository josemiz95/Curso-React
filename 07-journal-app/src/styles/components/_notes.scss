.notes__main-content{
	display: flex;
	flex-direction: column;
	height: 100%;
}

.notes__appbar{
	align-items: center;
	background-color: $primary;
	color: white;
	display: flex;
	justify-content: space-between;
	padding: 10px 20px;

	.btn{
		color: white;
	}
}

.notes__content{
	display: flex;
	flex-direction: column;
	height: 100%;
	padding: 20px;
}

.notes__title-input, .notes__textarea{
	border: none;
	color: $dark-grey;

	&:hover, &:focus{
		outline: none;
	}
}

.notes__title-input {
	font-size: 25px;
	font-weight: bold;
	margin-bottom: 20px;
}

.notes__textarea {
	font-size: 20px;
	flex: 1 1 auto;
	resize: none;
}

.notes__image img{
	height: 150px;
}