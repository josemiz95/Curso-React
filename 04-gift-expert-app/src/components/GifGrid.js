import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    const [images, setImages] = useState([]);
    useEffect(() => {
        getGift(category);
    }, []);
    

    const getGift = async (query) => {
        const apiKey = 'ECiPERL160ybBJ3RdhW1YHh0vWJwdwyp';
        const response = await fetch(`https://api.giphy.com/v1/gifs/search?limit=10&q=${query}&api_key=${apiKey}`);
        const {data} = await response.json();

        const gifs = data.map((gif) => {
            return {
                id: gif.id,
                title: gif.title,
                image: gif.images?.downsized_medium?.url
            }
        });

        setImages(gifs);
    }

    
  return (
    <>
        <h3>{category}</h3>
        <div className='card-grid'>
            {images.map(img => <GifGridItem key={img.id} { ...img }/>)}
        </div>
    </>
  )
}
