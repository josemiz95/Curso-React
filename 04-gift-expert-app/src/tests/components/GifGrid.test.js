import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid/>', () => {
    const category = 'One punch';

    test('debe mostrarse correctamente', () => { 
        useFetchGifs.mockReturnValue([ [], true ]);

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes', ()=>{
        const gifs = [{
            id: 'ABC',
            title: 'Titulo',
            image: 'https://algo.com'
        }];

        useFetchGifs.mockReturnValue([ gifs, false ]);
        
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});