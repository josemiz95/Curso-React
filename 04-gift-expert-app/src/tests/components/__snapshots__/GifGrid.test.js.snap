// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid/> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading
    </p>
  </div>
</Fragment>
`;
